lib/widgets/message_list.dart

```diff
@@ -459,30 +459,41 @@ class ScrollToBottomButton extends StatelessWidget {
   }
 }
 
-class MarkAsReadWidget extends StatelessWidget {
+class MarkAsReadWidget extends StatefulWidget {
   const MarkAsReadWidget({super.key, required this.narrow});
 
   final Narrow narrow;
 
+  @override
+  State<MarkAsReadWidget> createState() => MarkAsReadWidgetState();
+}
+
+class MarkAsReadWidgetState extends State<MarkAsReadWidget> {
```

Comment:
```
msglist: Handle loading state in MarkAsReadWidget

MarkAsReadWidget needs to be disabled during loading state to prevent
multiple requests to the server.

Additionally it is needed for styling/animation convenience.
```

That last sentence ("Additionally…") doesn't really make sense; let's drop it. (What is more "convenient" about disabling the button?) I think it was from a previous revision when this commit was focused more on an implementation detail rather than the changes that users could see.

Reply:
That's right. Thanks!